addons/stock_dropshipping/models/sale.py

```diff
@@ -42,6 +42,8 @@ def _compute_is_mto(self):
     def _get_qty_procurement(self, previous_product_uom_qty=False):
         # People without purchase rights should be able to do this operation
         purchase_lines_sudo = self.sudo().purchase_line_ids
+        if not any(purchase_lines_sudo.mapped(lambda pol: pol._is_dropshipped())):
+            return super()._get_qty_procurement(previous_product_uom_qty=previous_product_uom_qty)
         # We make sure that it's not a kit with dropshipped components
         if self.product_id == purchase_lines_sudo.product_id and purchase_lines_sudo.filtered(lambda r: r.state != 'cancel'):
```

Comment:
```suggestion
        # We make sure that it's not a kit with dropshipped components
        if any(purchase_lines_sudo.mapped(lambda pol: pol._is_dropshipped())) and self.product_id == purchase_lines_sudo.product_id and purchase_lines_sudo.filtered(lambda r: r.state != 'cancel'):
```
It's enough since we call super in the else branch.
Moreover, it's easier to read than `Not [Any/All]`